src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,98 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import { getNumbers } from '../../utils';
+
+interface Props {
+  firstItemOnPage: number,
+  lastItemOnPage: number,
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => {},
+}
+
+export const Pagination: React.FC<Props> = ({
+  firstItemOnPage,
+  lastItemOnPage,
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+  const pageArr = getNumbers(1, pagesAmount);
+
+  const allItems = getNumbers(1, total);
+
+  const itemsOnPageArr
+  = [...allItems].slice(firstItemOnPage - 1, lastItemOnPage);
+
+  const eventHandler = (newPage: number) => {
```

Comment:
```suggestion
  const eventHandler = (newPage: number) => () => {
```
better to return arrow function here and in jsx just call